### ISSUE TITLE
2 issues

### ISSUE THREAD

**Issue by fabznn** [blizzhackers/kolbot#363]
Hello!

I explored 2 issues which i cannot solve by myself.

first one:
my druid keeps stacking gems until its inventory full. after that he cant pick anything else.
pickit settings are ok, same as the other chars.
Why does he do that?
Same with my barb - he stacks various items only in his inventory, until inventory is full. he dont sell and he dont put anything into chest.

second one:
when i use "tele switch" option on my barb (zerker) - after teleporting he doesnt switch to his primary slot.
he attacks with his Hoto's which makes no dmg at all.
tried to set primary slot and also left it on auto detect.
nothing works properly.

any solutions?
greetings

**Comment by theBGuy** [comment]
teleswitch was never fully completed, you could add a weaponswitch check inside of `ClassAttack.doAttack`.

I think the stacking gems was a result of the recent GemHunter script

**Comment by theBGuy** [comment]
Stacking gems is now fixed by #361 
I'll look into making tele switch more viable

**Comment by fabznn** [comment]
alright, thank you very much! good to know. sadly i got a next issue, i will open another topic :-)

**Comment by icommitdesnet** [comment]
> I think the stacking gems was a result of the recent GemHunter script

yes. sorry. my bad: default config changed the behavior even without having gemhunting activated. default config must not enable anything here.
ill do a rework of the gemhuner next days according to theBGuys comments. mainly checking for available gems and getting one from stash before picking a shrine.
